src/coreclr/gc/gc.cpp

```diff
@@ -48186,6 +48186,42 @@ HRESULT GCHeap::Initialize()
     {
         return E_OUTOFMEMORY;
     }
+
+    /*
+     * Allocation requests less than loh_size_threshold will be allocated on the small object heap.
+     * For regions, we constrainted that objects must stay on a single region, so it is impossible to allocate
+     * an object larger than the available space on a single region.
+     *
+     * Note that that available space on the region is not the region size, but a bit smaller.
+     * The adjustment formula is based on these three implementation details:
+     *
+     * 1.) heap_segment_mem is set to the new pages + sizeof(aligned_plug_and_gap) in make_heap_segment,
+     *     effectively making the bytes before that unusable for objects.
+     * 2.) a_fit_segment_end_p set pad to Align(min_obj_size, align_const), effectively making some end bytes
+     *     unusable
+     * 3.) a_size_fit_p requires the available space to be larger than the allocated size + Align(min_obj_size, align_const)
+     *     again, effectively making some end bytes unusable.
+     *
+     * It is guaranteed that allocation request with exactly this amount or less will either succeed or fail (because we either
+     * cannot get new region or cannot commit memory), but never run into an infinite loop.
```

Comment:
```suggestion
     * It is guaranteed that an allocation request with this amount or less will succeed unless
     * we cannot commit memory for it.
```
if you can't a new region you wouldn't even be doing any check to see if the size can fit so we don't need to say "if we cannot get new region". succeeding also implies we cannot run into an infinite loop